source/slang/slang-ir-undo-param-copy.cpp

```diff
@@ -0,0 +1,141 @@
+#include "slang-ir-undo-param-copy.h"
+
+#include "slang-ir-dce.h"
+#include "slang-ir-insts.h"
+#include "slang-ir.h"
+
+#include <stdio.h> // For fprintf
+
+namespace Slang
+{
+// This pass transforms variables decorated with TempCallArgVarDecoration
+// by replacing them with direct references to the original parameters.
+// This is important for CUDA/OptiX targets where functions like 'IgnoreHit'
+// can prevent copy-back operations from executing.
+struct UndoParameterCopyVisitor
+{
+    IRBuilder builder;
+    IRModule* module;
+    bool changed = false;
+
+    UndoParameterCopyVisitor(IRModule* module)
+        : module(module)
+    {
+        builder.setInsertInto(module);
+    }
+
+    // Map from temporary variables (IRVar) to original parameter pointers
+    Dictionary<IRInst*, IRInst*> tempVarToOriginalParamPtrMap;
+
+    // Track instructions to remove (temp vars and their initializing stores)
+    List<IRInst*> instsToRemove;
+
+    // Process the entire module
+    void processModule()
+    {
+        // Process all functions in the module
+        for (auto inst = module->getModuleInst()->getFirstChild(); inst; inst = inst->getNextInst())
+        {
+            if (auto func = as<IRFunc>(inst))
+            {
+                processFunc(func);
+            }
+        }
+    }
+
+    // Process a single function
+    void processFunc(IRFunc* func)
+    {
+        tempVarToOriginalParamPtrMap.clear();
+        instsToRemove.clear();
+
+        // Pass 1: Identify TempCallArgVars and map them to their original parameter\'s pointer.
+        // Also, collect the temp var and its initializing store for removal.
+        for (auto block = func->getFirstBlock(); block; block = block->getNextBlock())
+        {
+            for (auto inst = block->getFirstInst(); inst; inst = inst->getNextInst())
+            {
+                if (auto varInst = as<IRVar>(inst))
+                {
+                    if (varInst->findDecoration<IRTempCallArgVarDecoration>())
+                    {
+                        IRStore* initializingStore = nullptr;
+                        IRInst* originalParamPtr = nullptr;
+
+                        // Scan for the store that initializes this varInst
+                        // This store should be in the same block, after varInst.
+                        // The value stored should be an IRLoad from the original parameter pointer.
+                        for (auto scanInst = varInst->getNextInst();
```

Comment:
Instead of traversing instructions following `varInst`, you should traverse its uses:

```
for (auto use = varInst->firstUse; use; use = use->nextUse)
{
     auto user = use->getUser();
     if (auto storeInst == as<IRStore>(user))
     {
              if (storeInst->getPtr() == varInst)
                 // we found the store into `varInst`.
     }
}
```